# <u>Sawasdee Dashboard Follow up (December 1st 2021 to January 31st 2022)</u>

## Checking for duplicates in each table

In [9]:
SELECT id,orderID, count(*) as 'Count'
FROM dbo.OrderPayment
GROUP BY id,orderId
HAVING COUNT(*) > 1

SELECT id, count(*) as 'Count'
FROM dbo.OrderList
GROUP BY id
HAVING COUNT(*) > 1

(0 rows affected)

(0 rows affected)

Total execution time: 00:00:00.009

id orderID Count

id Count

There are no duplicate id's or orderIDs in the OrderPayment table

There are also no duplicate id's in the OrderList table

## Looking through each table

In [17]:
select top 5 id,OrderId,TotalAmount,FaceAmount,GratuityAmount,TenderAmount,Change,PaidMethodName,CCVendor,CreatedBy,CreatedWhen
from dbo.OrderPayment
where CreatedWhen 
BETWEEN CONVERT(datetime,'2021-12-01') AND CONVERT(datetime, '2022-01-31 23:59:59')

select top 5 id,OrderNo,SubTotal,Tax,Total,NumberofGuest,CreatedWhen,OrderType,CreatedBy,TableId 
from dbo.OrderList
where CreatedWhen 
BETWEEN CONVERT(datetime,'2021-12-01') AND CONVERT(datetime, '2022-01-31 23:59:59')

select * from dbo.OrderType

(5 rows affected)

(5 rows affected)

(7 rows affected)

Total execution time: 00:00:00.082

id OrderId TotalAmount FaceAmount GratuityAmount TenderAmount Change PaidMethodName CCVendor CreatedBy CreatedWhen 50261943245780518 50261943239167508 14.1500 14.1500 3.0000 17.1500 0.0000 Credit Card VISA 1003 2021-12-02 12:00:45.780 50261948015474336 50261947525481767 14.1500 14.1500 0.0000 20.0000 5.8500 Cash 1003 2021-12-02 13:20:15.473 50261950855829811 50261947084503598 75.6600 75.6600 13.6200 89.2800 0.0000 Credit Card VISA 1003 2021-12-02 14:07:35.830 50261952035971301 50261949788196830 58.8300 58.8300 11.1700 70.0000 0.0000 Credit Card VISA 1003 2021-12-02 14:27:15.970 50261954685786069 50261951127813504 51.1800 51.1800 10.2400 61.4200 0.0000 Credit Card VISA 1003 2021-12-02 15:11:25.787

id OrderNo SubTotal Tax Total NumberofGuest CreatedWhen OrderType CreatedBy TableId 50261943239167508 1 12.9500 1.2000 14.1500 0 2021-12-02 12:00:39.183 4 1003 0 50261947084503598 3 69.2500 6.4100 75.6600 2 2021-12-02 13:04:44.527 1 1003 19 50261947525481767 4 12.9500 1.2000 14.1500 0 2021-12-02 13:12:05.513 3 1003 0 50261949788196830 6 53.8500 4.9800 58.8300 2 2021-12-02 13:49:48.197 1 1003 7 50261951127813504 7 46.8500 4.3300 51.1800 2 2021-12-02 14:12:07.830 1 1003 11

id AliasName AliasNameAlt AliasNameAlt1 IsEnable IsQuickService IsEnterNoGuest PriceSet1 PriceSet2 PriceSet3 PriceSet4 PriceSet5 EventSet1Id EventSet2Id EventSet3Id EventSet4Id EventSet5Id MenuSet1Id MenuSet2Id MenuSet3Id MenuSet4Id MenuSet5Id IsAutoCharge MinGuestToAutoCharge DefaultCharge IsServiceCharge IsChargeAfterTax IsChargeBeforerDiscount IsExemptTax1 IsExemptTax2 IsExemptTax3 IsEnableTable IsAllowNotReady AutoReadyTime DeliveryChargeDefaultAmt CompensationDefaultAmt IsTurnOffDispatcher IsPromptEnterCustName PromptAliasName SysTimeStamp ReceiptFooter DonePrintOpt DonePrintCopy PayPrintOpt PayPrintCopy IsUseServerBank IsByPassPhoneNo ChargeAliasName PackagerPrintOpt PackagerPrintCopy DefaultAmountCharge IsPercentAutoCharge IsEnablePreAuth TabPrintOpt TabPrintCopy DefaultPreAuthAmount IsEnableTab IsAllowChangeAuthAmount IsEnableFastTab VoidPreAuthPrintOpt VoidPreAuthPrintCopy IsForcedPay DefaultDiscountID 1 Dining 1 0 1 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 1 6.0000 18.0000 1 0 1 0 0 0 1 1 0.0000 0.0000 0.0000 0 0 Enter Information T202103302146320000 THANK YOU!!!
 0 1 0 1 0 0 1 1 0.0000 1 0 0 1 0.0000 1 0 0 1 1 0 0 2 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0.0000 0.0000 0 0 0 0 0 0 0 0 0.0000 0.0000 0.0000 1 0 T202201181034580120 THANK YOU!!!
 1 1 0 1 1 0 1 1 0.0000 1 0 0 1 0.0000 1 0 0 1 1 0 0 3 TAKE OUT 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0.0000 0.0000 1 0 0 0 0 0 0 0 0.0000 0.0000 0.0000 0 0 T201905291709461190 THANK YOU!!!
 1 1 0 1 0 0 1 1 0.0000 1 0 0 1 0.0000 1 0 0 1 1 0 0 4 WALK IN 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0.0000 0.0000 1 0 0 0 0 0 0 0 0.0000 0.0000 0.0000 0 1 Enter Customer Name T201905291709422863 THANK YOU!!!
 1 1 0 1 0 0 1 1 0.0000 1 0 0 1 0.0000 1 0 0 1 1 0 0 5 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0.0000 0.0000 0 0 0 0 0 0 0 0 0.0000 0.0000 0.0000 0 0 T201905211002036016 1 1 1 1 0 0 0 1 0.0000 1 0 0 1 0.0000 1 0 0 1 1 0 0 62 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0.0000 0.0000 0 0 0 0 0 0 0 1 0.0000 0.0000 0.0000 1 0 T202201181034580120 THANK YOU!!!
 1 1 0 1 1 0 1 1 0.0000 1 0 0 1 0.0000 1 0 0 1 1 0 0 63 TAKE OUT 1 0 0 0 0 0 0 0 0 0 0 0 0 1 0 0 0 0 0 0.0000 0.0000 1 0 0 0 0 0 0 1 0.0000 0.0000 0.0000 0 0 T201905291709461190 THANK YOU!!!
 1 1 0 1 0 0 1 1 0.0000 1 0 0 1 0.0000 1 0 0 1 1 0 0

OrderId from OrderPayment table corresponds to Id from the OrderList table. This will be important when we need to check what order type each order is. FaceAmount from the OrderPayment table is used when people split a check. It will be useful when we sum up the sales. Whether or not customers use Cash or Credit Card most likely will not be a good indicator of any business decisions, therefore we won't be focusing on it.

The OrderType table will be used to check the names for each order type. We can join this table with the OrderList table to computationally what the type of order is. A problem with this table is that there are three blank AliasNames and there's two id's for 'TAKE OUT'. We're going to have to figure out what these are.

Later found: 

OrderList.OrderType names by number are: TakeOut ==  '63, 3', WalkIn = Order Type '4', Dining == '1', Online Delivery == '62'

There are two OrderType '5' made in early 2020 but hasn't been seen since. It is most likely some type of test order made by our tech support

## Breaking down the number of orders by order types

In [39]:
WiTH
	OnlineOrderCount
	AS
	(
		select count(*) as 'Online Pick-up Count'
		from dbo.OrderPayment
			join dbo.OrderList on
	OrderPayment.OrderId = OrderList.id
		where OrderPayment.CreatedWhen 
	BETWEEN CONVERT(datetime,'2021-12-01') AND CONVERT(datetime, '2022-01-31 23:59:59')
			and PaidMethodName = 'Online Payment'
			and OrderList.OrderType = '63'
	),

	CountOrderTypes
	AS
	(
		select
			count(*) as 'Order Count',
			OrderType.AliasName as Name,
			OrderDetails.OrderType as 'Order Type'
		from dbo.OrderList as OrderDetails
			inner join dbo.OrderType on
	OrderDetails.OrderType = OrderType.id
			inner join dbo.OrderPayment on
	OrderPayment.OrderId = OrderDetails.id
		where OrderPayment.CreatedWhen 
	BETWEEN CONVERT(datetime,'2021-12-01') AND CONVERT(datetime, '2022-01-31 23:59:59')
		group by OrderDetails.OrderType, OrderType.AliasName, OrderDetails.OrderType
	)

select
	SUM(CASE WHEN Name = 'TAKE OUT' or Name = 'Walk In' THEN "Order Count" ELSE 0 END) - OnlineOrderCount.[Online Pick-up Count] as 'Combined Take Out',
	SUM(CASE WHEN Name = 'Dining' THEN "Order Count" ELSE 0 END) as 'Dining',
	SUM(CASE WHEN Name = '' THEN "Order Count" ELSE 0 END) as 'Online Delivery',
	[Online Pick-up Count] as 'Online Pick-Up Orders'
from CountOrderTypes, OnlineOrderCount
group by [Online Pick-up Count];

(1 row affected)

Total execution time: 00:00:01.037

Combined Take Out Dining Online Delivery Online Pick-Up Orders 738 1001 22 278

Using the same two CTEs, we can write a statement that takes half the time a different way using union and subquerying in the 'FROM' clause.

In [38]:
WITH
    OnlineOrderCount
    AS
    (
        select count(*) as 'Online Pick-up Count'
        from dbo.OrderPayment
            join dbo.OrderList on
	OrderPayment.OrderId = OrderList.id
        where OrderPayment.CreatedWhen 
	BETWEEN CONVERT(datetime,'2021-12-01') AND CONVERT(datetime, '2022-01-31 23:59:59')
            and PaidMethodName = 'Online Payment'
            and OrderList.OrderType = '63'
    ),

    CountOrderTypes
    AS
    (
        select
            count(*) as 'Order Count',
            OrderType.AliasName as Name,
            OrderDetails.OrderType as 'Order Type'
        from dbo.OrderList as OrderDetails
            inner join dbo.OrderType on
	OrderDetails.OrderType = OrderType.id
            inner join dbo.OrderPayment on
	OrderPayment.OrderId = OrderDetails.id
        where OrderPayment.CreatedWhen 
	BETWEEN CONVERT(datetime,'2021-12-01') AND CONVERT(datetime, '2022-01-31 23:59:59')
        group by OrderDetails.OrderType, OrderType.AliasName, OrderDetails.OrderType
    )
select sum("Combined Take Out" - "Online Pick-up Count") as 'Combined Take Out', sum("Dining") as 'Dining',
    sum("Online Delivery") as 'Online Delivery', sum("Online Pick-up Count") as 'Online Pick-up Count'
from (
        select
            SUM(CASE WHEN Name = 'TAKE OUT' or Name = 'Walk In' THEN "Order Count" ELSE 0 END) as 'Combined Take Out',
            SUM(CASE WHEN Name = 'Dining' THEN "Order Count" ELSE 0 END) as 'Dining',
            SUM(CASE WHEN Name = '' THEN "Order Count" ELSE 0 END) as 'Online Delivery',
            0 as 'Online Pick-up Count'
        from CountOrderTypes
    union
        select 0, 0, 0, [Online Pick-up Count]
        from OnlineOrderCount) unionsubquery

(1 row affected)

Total execution time: 00:00:00.027

Combined Take Out Dining Online Delivery Online Pick-up Count 738 1001 22 278

Notes: 

I subtracted the number of online pickup from the total take out orders and created a new column for it since order type 63 shares both regular take out orders and online pick-up orders

OrderList includes online orders as take out, so we must look at the OrderPayment table to get the number of Online Orders.  

Results:

Order from most popular to least popular: Dining(1001) \> Walk In + Both 'Take Out's (738) \> Online Pick-Up (278) \> Online Delivery (22)

The combination of take out orders and online pick up orders has now superseded dining as the most popular ordering method.

## Finding tip amounts and percentages by order types

In [54]:
select cast(sum(case when [Avg Online Pick-up Tips] is not null then t1.[Avg Online Pick-up Tips] Else 0 END) as decimal(18,2)) as 'Avg Online Pick-up Tips' , --cast as deciminal is used to truncate decimals past 2 places
    cast(sum(case when [Avg Online Pick-up Tips Percentage] is not null then [Avg Online Pick-up Tips Percentage] Else 0 END) as decimal(18,2)) as 'Avg Online Pick-up Tips Percentage',
    cast(sum(case when [Take Out Avg Tip Amount] is not null then [Take Out Avg Tip Amount] Else 0 END) as decimal(18,2))  as 'Take Out Avg Tip Amount',
    cast(sum(case when [Take Out Avg Tip Percentage] is not null then [Take Out Avg Tip Percentage] Else 0 END) as decimal(18,2))  as 'Take Out Avg Tip Percentage',
    cast(sum(case when [Dine In Avg Tip Amount] is not null then [Dine In Avg Tip Amount] Else 0 END) as decimal(18,2))  as 'Dine In Avg Tip Amount',
    cast(sum(case when [Dine In Avg Tip Percentage] is not null then [Dine In Avg Tip Percentage] Else 0 END) as decimal(18,2)) as 'Dine In Avg Tip Percentage'
from (
	select ROUND(sum(GratuityAmount)/Count(*),2) as 'Avg Online Pick-up Tips'
    from dbo.OrderPayment join
        dbo.OrderList on
		OrderPayment.OrderId = OrderList.id
    where OrderPayment.CreatedWhen 
		BETWEEN CONVERT(datetime,'2021-12-01') AND CONVERT(datetime, '2022-01-31 23:59:59')
        and OrderPayment.PaidMethodName = 'Online Payment'
        and OrderList.OrderType = '63') as t1
    full outer join (
	select ROUND((sum(GratuityAmount)/sum(FaceAmount)*100),2) as 'Avg Online Pick-up Tips Percentage'
    from dbo.OrderPayment join
        dbo.OrderList on
		OrderPayment.OrderId = OrderList.id
    where OrderPayment.CreatedWhen 
		BETWEEN CONVERT(datetime,'2021-12-01') AND CONVERT(datetime, '2022-01-31 23:59:59')
        and OrderPayment.PaidMethodName = 'Online Payment'
        and OrderList.OrderType = '63') as t2
    on t1.[Avg Online Pick-up Tips] = t2.[Avg Online Pick-up Tips Percentage]
    full outer join (
	SELECT ROUND(sum(Orders.GratuityAmount)/COUNT(*),2) as 'Take Out Avg Tip Amount'
    FROM dbo.OrderPayment AS Orders
        INNER JOIN dbo.OrderList AS OrderDetails
        on Orders.OrderId = OrderDetails.id
    WHERE Orders.CreatedWhen 
		BETWEEN CONVERT(datetime,'2021-12-01') AND CONVERT(datetime, '2022-01-31 23:59:59')
        AND OrderDetails.OrderType IN (63,4,3)
        and Orders.PaidMethodName != 'Online Payment') as t3
    on t2.[Avg Online Pick-up Tips Percentage] = t3.[Take Out Avg Tip Amount]
    full outer join (
	SELECT ROUND((sum(Orders.GratuityAmount)/sum(Orders.FaceAmount)*100),2) as 'Take Out Avg Tip Percentage'
    FROM dbo.OrderPayment AS Orders
        INNER JOIN dbo.OrderList AS OrderDetails
        on Orders.OrderId = OrderDetails.id
    WHERE Orders.CreatedWhen 
		BETWEEN CONVERT(datetime,'2021-12-01') AND CONVERT(datetime, '2022-01-31 23:59:59')
        AND OrderDetails.OrderType IN (63,4,3)
        and Orders.PaidMethodName != 'Online Payment') as t4
    on t4.[Take Out Avg Tip Percentage] = t3.[Take Out Avg Tip Amount]
    full outer join (
	SELECT ROUND(sum(Orders.GratuityAmount)/COUNT(*),2) as 'Dine In Avg Tip Amount'
    FROM dbo.OrderPayment AS Orders
        INNER JOIN dbo.OrderList AS OrderDetails
        on Orders.OrderId = OrderDetails.id
    WHERE Orders.CreatedWhen 
		BETWEEN CONVERT(datetime,'2021-12-01') AND CONVERT(datetime, '2022-01-31 23:59:59')
        AND OrderDetails.OrderType = 1) t5
    on t4.[Take Out Avg Tip Percentage] = t5.[Dine In Avg Tip Amount]
    full outer join (
	SELECT ROUND((sum(Orders.GratuityAmount)/sum(Orders.FaceAmount)*100),2) as 'Dine In Avg Tip Percentage'
    FROM dbo.OrderPayment AS Orders
        INNER JOIN dbo.OrderList AS OrderDetails
        on Orders.OrderId = OrderDetails.id
    WHERE Orders.CreatedWhen 
		BETWEEN CONVERT(datetime,'2021-12-01') AND CONVERT(datetime, '2022-01-31 23:59:59')
        AND OrderDetails.OrderType = 1) as t6
    on t5.[Dine In Avg Tip Amount] = t6.[Dine In Avg Tip Percentage]

(1 row affected)

Total execution time: 00:00:00.095

Avg Online Pick-up Tips Avg Online Pick-up Tips Percentage Take Out Avg Tip Amount Take Out Avg Tip Percentage Dine In Avg Tip Amount Dine In Avg Tip Percentage 4.41 9.38 2.81 7.88 5.97 10.16

Notes: 

Take out in this case does not include online pick-up orders.

Results: 

Order from highest to lowest average tips: Dine In \> Online Pick-Up  \> Take Out

The order of absolute values agree with the order of percentages (there isn't a big gap in order sales between each category)

## Finding the amount saved per month by online delivery orders

In [36]:
select
    sum(FaceAmount)*0.25 as 'Amount Saved'
from dbo.OrderList as OrderDetails
    inner join dbo.OrderPayment on
	OrderPayment.OrderId = OrderDetails.id
where OrderPayment.CreatedWhen 
	BETWEEN CONVERT(datetime,'2021-12-01') AND CONVERT(datetime, '2022-01-31 23:59:59')
    and OrderType = '62'
group by OrderDetails.OrderType, OrderDetails.OrderType

(1 row affected)

Total execution time: 00:00:00.036

Amount Saved 339.260000

Results:

There are about $1360 in online delivery sales. We expect to save about 25% of that when people use our website instead of Doordash or UberEats. There is a 30% cut and we have a cancellation rate between 2% and 5%. We were able to save about $340($1360\*25%) in December and January off of the online delivery orders.

# Conclusion and Insight

Restaurant servers get the most tips from dine in orders at an average of 10.16%. Online pick-up orders are the second to highest order type our servers get at an average of 9.38% per order. Call in take out orders and walk in orders come in last at a low average of 7.88%. We should try to convert as many people as possible to use online pick-up orders. Having customers use our online services is much faster for our servers and they have a higher average tip rate. I believe this is due to the fact we have selectable tip amounts on our website. People are more likely to tip if they are shown preselected options. At this restaurant, we were able to save about $340 since we've had the new online delivery system. There is about a 50%-50% split between take out orders(including online orders) and dine in orders. Before, in August 2021 and September 2021 we've seen that the sales were more centered around dine in orders at about a 60%-40% split. This may be because this analysis is in the middle of winter and people are more likely to order take out or order online when it is cold. We expect more dine in sales when spring arrives and the Boardwalk opens up on weekdays.